src/StructuralModel/StructuralBoundaryConditions.jl

```diff
@@ -289,23 +289,26 @@ function Base.replace!(sb::StructuralBoundaryCondition,
 
     # node boundary conditions replacement
     node_boundary_conditions = node_bcs(sb)
-    if haskey(node_boundary_conditions, old_bc)
+    node_bc_type = eltype(keys(node_boundary_conditions))
+    if (old_bc isa node_bc_type) && haskey(node_boundary_conditions, old_bc)
```

Comment:
why is the new clause needed?

Reply:
Ah okay, the thing here si that parametric type of the `BCtoEntitiy{BC,N} ` can be defined initially only with `FixedDof` boundary condition. Then the `haskey` method throws an error if the key that you are checking is a different type. This sth that u have to keep in mind if a BC replacement is done before defining the `StructuralBoundaryCondition` struct.